backend/src/impl/default_controllers_impl.py

```diff
@@ -150,12 +160,52 @@ def benchmarkconfigs_get(parent: Optional[str]) -> list[BenchmarkConfig]:
 def benchmark_benchmark_id_get(benchmark_id: str) -> Benchmark:
     config = BenchmarkConfig.from_dict(BenchmarkUtils.config_dict_from_id(benchmark_id))
     if config.type == "abstract":
-        return Benchmark(config, None)
-    sys_infos = BenchmarkUtils.load_sys_infos(config)
-    orig_df = BenchmarkUtils.generate_dataframe_from_sys_infos(config, sys_infos)
-    view_dfs = BenchmarkUtils.generate_view_dataframes(config, orig_df)
-    views = [BenchmarkUtils.dataframe_to_table(k, v) for k, v in view_dfs]
-    return Benchmark(config, views)
+        return Benchmark(config, None, None)
+    file_path = benchmark_id + "_benchmark.json"
+    plot_path = benchmark_id + "_plot.csv"
+    benchmark_file = open_cached_file(file_path, datetime.timedelta(days=1))
+    if not benchmark_file:
+        sys_infos = BenchmarkUtils.load_sys_infos(config)
+        orig_df = BenchmarkUtils.generate_dataframe_from_sys_infos(config, sys_infos)
+        view_dfs = BenchmarkUtils.generate_view_dataframes(config, orig_df)
+        json_dict = {k: v.to_dict() for k, v in view_dfs}
+        benchmark_file = os.path.join(get_cache_dir(), sanitize_path(file_path))
+        with open(benchmark_file, "w") as outfile:
+            json.dump(json_dict, outfile)
+        if config.parent:
+            list_data = [
+                json_dict["Demographic-weighted Global Average"]["score"][0],
+                json_dict["Linguistic-weighted Global Average"]["score"][0],
+                datetime.datetime.now(),
+            ]
+            plot_path = os.path.join(get_cache_dir(), sanitize_path(plot_path))
+            with open(plot_path, "a", newline="") as f_object:
+                writer_object = writer(f_object)
+                writer_object.writerow(list_data)
+                f_object.close()
+    update_time = str(datetime.datetime.fromtimestamp(os.path.getmtime(benchmark_file)))
+    f = open(benchmark_file)
+    views = [
+        BenchmarkUtils.dataframe_to_table(k, pd.DataFrame.from_dict(v))
+        for k, v in json.load(f).items()
+    ]
+    return Benchmark(config, views, update_time)
+
+
+def benchmark_plot_benchmark_id_get(benchmark_id):
+    config = BenchmarkConfig.from_dict(BenchmarkUtils.config_dict_from_id(benchmark_id))
+    if config.type == "abstract" or not config.parent:
+        return PlotData([], [], [])
+    plot_path = os.path.join(get_cache_dir(), benchmark_id + "_plot.csv")
+    demographic_weighted_list, linguistic_weighted_list, times = [], [], []
+    with open(plot_path, "r", newline="") as f_object:
+        reader_object = reader(f_object)
+        for row in reader_object:
+            demographic_weighted_list.append(float(row[0]))
+            linguistic_weighted_list.append(float(row[1]))
+            times.append(datetime.datetime.strptime(row[2], "%Y-%m-%d %H:%M:%S.%f"))
```

Comment:
```suggestion
            times.append(datetime.datetime.strptime(row[2], "%Y-%m-%d %H:%M:%S"))
```

Reply:
Unfortunately, when I tried to do this, it threw me an error like "ValueError: unconverted data remains: .293000" :(